src/components/AsyncTable/AsyncTable.tsx

```diff
@@ -124,18 +132,25 @@ export const AsyncTable = <T extends unknown>({
           canSwitchViews={Boolean(gridComponent) && Boolean(rowComponent)}
           onSwitchViews={(isGridView: boolean) => setIsGridView(isGridView)}
           isGridView={isGridView}
+          searchPlaceholder={searchKey?.name && `Search by ${searchKey?.name}`}
           searchQuery={searchQuery}
           onChangeSearchQuery={setSearchQuery}
           isSearchable={Boolean(searchKey)}
         />
       )}
-      <InfiniteScrollWrapper className={className}>
-        {isGridView ? (
-          <Grid cards={chunkedComponents} isSingleColumnGrid={isSingleColumnGrid} />
-        ) : (
-          <Table rows={chunkedComponents} columns={columns} isLoading={isLoading} />
-        )}
-      </InfiniteScrollWrapper>
+      {query && !isLoading && filteredData.length ? (
```

Comment:
can we use another variable here?

```
const hasItems = query && !isLoading && filteredData.length;
```

Hmmm, by the way, if `query` is empty we show empty message? -> `query && ... ?  <table> : <empty>`?